test/user.test.js

```diff
@@ -0,0 +1,54 @@
+const dotenv = require('dotenv');
+const request = require('supertest');
+const { appDataSource } = require('../models/data-source');
+const { createApp } = require('../app');
+dotenv.config();
+
+jest.mock('../util/axios');
+const user = require('../util/axios');
+
+describe('social-login', () => {
+  let app;
+
+  beforeAll(async () => {
+    app = createApp();
+    await appDataSource.initialize();
+  });
+  afterAll(async () => {
+    await appDataSource.query(`SET foreign_key_checks = 0`);
+    await appDataSource.query(`TRUNCATE users`);
+    await appDataSource.query(`SET foreign_key_checks = 1`);
+    await appDataSource.destroy();
+  });
+
+  test('FAILED: NEED_ACCESS_TOKEN', async () => {
+    await request(app)
+      .post('/users/login')
+      .expect(400)
+      .expect({ message: 'NEED_ACCESS_TOKEN' });
+  });
+
+  test('SUCCESS: BODY_WITH_TOKEN', async () => {
+    user.axiosUser.mockReturnValue(
+      Promise.resolve({
+        data: {
+          id: 12124124,
+          properties: {
+            profile_image: 'skldjkalsdjkladjsk',
+            name: 'hihihi',
+          },
+          kakao_account: {
+            email: 'asdadsasd@sadasdasds',
+          },
+        },
+      })
+    );
```

Comment:
이 방법도 괜찮지만 axios 모듈이 반환하는 결과값 자체를 mocking 하면 됩니다.

아래 방법대로 한다면 실제 app에서 사용되는 axios 함수 결과 값이 아래의 mockReturnValue()로 치환되기 때문에 토큰에 어떤 값을 넣어도 문제가 되지 않습니다.

```suggestion
    axios.get = jest.fn().mockReturnValue({
      data: {
        id: 123456789,
        connected_at: "2022-12-30T109:00:00Z",
        properties: {
          nickname: "test",
        },
        kakao_account: {
          profile_nickname_needs_agreement: false,
          profile: {
            nickname: "test",
          },
          has_email: true,
          email_needs_agreement: false,
          is_email_valid: true,
          is_email_verified: true,
          email: "test@test.com",
        },
      },
    })
```